src/util/prolog/prologStringManipulation.ts

```diff
@@ -1,41 +1,122 @@
-const fetchFileExtension = /(.*)\.[^\.]*/;
-const fetchModuleDeclaration = /:-( |)module\([^)]*\) *\.[\r\t\f\v ]*\n?/g;
-const fetchPartialLineCommentsRegExp = /(^[^\n%]+)%[^\n\r]*(\r?\n)/gm;
-const fetchFullLineCommentsRegExp = /^[\r\t\f\v ]*%[^\n]*\n/gm;
-const fetchEmptyLinesRegExp = /\n(?:\s)*\n/gm;
-const fetchDoubleWhitespacesRegExp = /  */gm;
+import {fetchAllMatchesForAGroup} from "../regExpUtils";
+import builtInPredicates from "./builtInPredicates";
 
 export function removeFileExtension(path: string) {
+	const fetchFileExtension = /(.*)\.[^\.]*/;
+
 	const fetch: string[] | null = fetchFileExtension.exec(path);
 	if (fetch === null) { return ""; }
 	return fetch[1];
 }
 
 export function removeModuleDeclarations(prologProgram: string) {
+	const fetchModuleDeclaration = /:-( |)module\([^)]*\) *\.[\r\t\f\v ]*\n?/g;
+
 	return prologProgram.replace(fetchModuleDeclaration, "");
 }
 
 function removeFullLineComments(prologProgram: string) {
+	const fetchFullLineCommentsRegExp = /^[\r\t\f\v ]*%[^\n]*\n/gm;
+
 	return prologProgram.replace(fetchFullLineCommentsRegExp, "");
 }
 
 function removePartialLineComments(prologProgram: string) {
+	const fetchPartialLineCommentsRegExp = /(^[^\n%]+)%[^\n\r]*(\r?\n)/gm;
```

Comment:
For me it is not totally clear what a partial line comment is. Could you give an example?

Reply:
With those I meant comments like this:
```
somefunction(x, y). // This is the partial line comment. It doesn't start at the beginning of the line.
```